src/app/category.service.ts

```diff
@@ -0,0 +1,13 @@
+import { Injectable } from '@angular/core';
+import { AngularFireDatabase } from '@angular/fire/database';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CategoryService {
+  constructor(private db: AngularFireDatabase) {}
+
+  getCategories() {
+    return this.db.list('/categories', (ref) => ref.orderByChild('name'));
```

Comment:
I would do all the logic to pull everything out and format the response to just a category[] in here